lib/xapi-stdext-pervasives/pervasiveext.ml

```diff
@@ -36,6 +36,24 @@ let finally fct clean_f =
   clean_f ();
   result
 
+let finally_with_exception fct clean_f =
```

Comment:
See the long discussion about `finally` here: https://github.com/ocaml/ocaml/pull/1855/files and the solution that went into OCaml 4.08 (I believe). It might be best to start from there when trying to improve `finally`. I don't mind the name but more the details of the mechanics.

Reply:
See also discussion in https://github.com/ocaml/ocaml/pull/2118#issuecomment-452747808

Reply:
after reading those comments and edvin's comment on my other PR, i shouldn't be using `finally` in my use case as it mplies the wrong semantics, so i'll close this